rackcity/views/decommission_views.py

```diff
@@ -119,20 +126,42 @@ def decommission_asset(request):
                 },
                 status=HTTPStatus.BAD_REQUEST,
             )
-        if not user_has_asset_permission(request.user, site=asset.rack.datacenter):
+        # TODO: permissions for blades
+        if asset.rack and not user_has_asset_permission(
+            user, site=asset.rack.datacenter
+        ):
             return JsonResponse(
                 {
                     "failure_message": Status.AUTH_ERROR.value
                     + AuthFailure.ASSET.value,
                     "errors": "User "
-                    + request.user.username
+                    + user.username
                     + " does not have asset permission in datacenter id="
                     + str(asset.rack.datacenter.id),
                 },
                 status=HTTPStatus.UNAUTHORIZED,
             )
 
     if change_plan:
+        blades = AssetCP.objects.filter(chassis=decommissioned_asset_cp.id)
+        for blade in blades:
+            data_for_blade = data.copy()
+            data_for_blade["id"] = blade.id
+            try:
+                decommission_asset_parameterized(data_for_blade, query_params, user)
+                # Assume that if this call doesn't raise an exception, it was successful
+                # None of the failed responses above are possible
+            except Exception as error:
+                return JsonResponse(
+                    {
+                        "failure_message": Status.DECOMMISSION_ERROR.value
+                        + "Unable to decommission blade: '"
+                        + str(blade.asset_number)
+                        + "'. ",
+                        "errors": str(error),
+                    },
+                    status=HTTPStatus.BAD_REQUEST,
+                )
```

Comment:
since this logic is the same it might be nice to have it in a separate method and you either pass in `Asset` or `AssetCP` and do the whole thing on the model 😁

Reply:
You are right that might be nice